src/fingerprint_bond.cpp

```diff
@@ -0,0 +1,917 @@
+		r1 = r1-trunc(r1);
+		double *p0 = &expcuttable[(m1-1)*this->k];
+		double *p1 = &expcuttable[m1*this->k];
+		double *p2 = &expcuttable[(m1+1)*this->k];
+		double *p3 = &expcuttable[(m1+2)*this->k];
+		for (kk=0;kk<this->k;kk++){
+			expr[jj][kk] = p1[kk]+0.5*r1*(p2[kk]-p0[kk]+r1*(2.0*p0[kk]-5.0*p1[kk]+4.0*p2[kk]-p3[kk]+r1*(3.0*(p1[kk]-p2[kk])+p3[kk]-p0[kk])));
+		}
+		double* q = &dfctable[m1-1];
+		double* ri = &rinvsqrttable[m1-1];
+		double dfc = q[1] + 0.5 * r1*(q[2] - q[0] + r1*(2.0*q[0] - 5.0*q[1] + 4.0*q[2] - q[3] + r1*(3.0*(q[1] - q[2]) + q[3] - q[0])));
+		double rinvs = ri[1] + 0.5 * r1*(ri[2] - ri[0] + r1*(2.0*ri[0] - 5.0*ri[1] + 4.0*ri[2] - ri[3] + r1*(3.0*(ri[1] - ri[2]) + ri[3] - ri[0])));
+
+		expr[jj][p]=delx*rinvs;
+		expr[jj][p+1]=dely*rinvs;
+		expr[jj][p+2]=delz*rinvs;
+		//Hack to avoid nan when x y or z component of radial vector is exactly 0. Shouldn't affect accuracy.
+		if (expr[jj][p]*expr[jj][p]<0.000000000001){
+		  expr[jj][p] = 0.000001;
+		}
+		if (expr[jj][p+1]*expr[jj][p+1]<0.000000000001){
+		  expr[jj][p+1] = 0.000001;
+		}
+		if (expr[jj][p+2]*expr[jj][p+2]<0.000000000001){
+			expr[jj][p+2] = 0.000001;
+		}
+		expr[jj][p+3] = -dfc*expr[jj][p];
+		expr[jj][p+4] = rinvs/expr[jj][p];
+		expr[jj][p+5] = rinvs*expr[jj][p];
+		expr[jj][p+6] = -dfc*expr[jj][p+1];
+		expr[jj][p+7] = rinvs/expr[jj][p+1];
+		expr[jj][p+8] = rinvs*expr[jj][p+1];
+		expr[jj][p+9] = -dfc*expr[jj][p+2];
+		expr[jj][p+10] = rinvs/expr[jj][p+2];
+		expr[jj][p+11] = rinvs*expr[jj][p+2];
+	}
+
+	int kb = this->k;
+	int mb = this->m;
+	count = startingneuron;
+	double Bb[mb];
+	double dBbx;
+	double dBby;
+	double dBbz;
+//	double dBbx1[mb];
+//	double dBby1[mb];
+//	double dBbz1[mb];
+	double yprod;
+	for (mcount=0;mcount<countmb;mcount++){
+		int *M = Mf[mcount];
+		int *coeffx = this->coeffx[mcount];
+		int *coeffy = this->coeffy[mcount];
+		int *coeffz = this->coeffz[mcount];
+		int *coeff = this->coeff[mcount];
+		a = mb+1;
+		for (a1=0;a1<mb;a1++){
+			if (Mf[mcount][a1+1]==0){
+				a = a1;
+				break;
+			}
+		}
+		for (n=0;n<kb;n++){
+			for (a1=0;a1<mb;a1++){
+				Bb[a1]=0;
+//				dBbx1[a1] = 0;
+//				dBby1[a1] = 0;
+//				dBbz1[a1] = 0;
+			}
+			ai = n;
+			double y1 = alpha_k[ai]/re;
+			//loop over jtype to get Bb
+			for (jj=0;jj<jnum;jj++){
+				if (expr[jj][0]==0){continue;}
+//				j = jlist[jj];
+//				j &= NEIGHMASK;
+//				jtype = pair->map[type[j]];
+				jtype = tn[jj];
+				if (atomtypes[1] != nelements && atomtypes[1] != jtype){
+					continue;
+				}
+				double yprod = expr[jj][ai];
+				double *y4 = &expr[jj][p];
+				for (a2=0;a2<a;a2++){
+					yprod *= y4[M[a2+1]];
+				}
+				for (a2=a;a2<mb;a2++){
+					Bb[a2]=Bb[a2]+yprod;
+//					if (i==5 && a2 == 1 && n==0){
+//						char str[128];
+//						sprintf(str,"%f %f %f\n",Bb[a2],yprod,y4[M[a2]]);
+//						std::cout<<str;
+//					}
+					yprod *= y4[M[a2+1]];
+				}
+			}
+			if (atomtypes[1]!=atomtypes[2]){//Bb!=Bg
+				double Bg[mb];
+//				double dBgx;
```

Comment:
please remove all commented out code. if there is debug code that should remain, please place it into
```
#if DEBUG_RANN
// debug code for XXX
[...]
#endif
```
and then place a `#define DEBUG_RANN 0` in a prominent position at the beginning of the file.